pkg/resource/replication_group/custom_set_output.go

```diff
@@ -249,3 +264,31 @@ func (rm *resourceManager) provideEvents(
 	}
 	return events, nil
 }
+
+// to be called upon a successful creation or modification (but not read/describe),
+// setLastRequestedLogDeliveryConfigurations copies desired.Spec.LogDeliveryConfigurations
+// into the annotations of the object.
+// r represents desired, ko represents latest
+func (rm *resourceManager) setLastRequestedLogDeliveryConfigurations(
+	r *resource,
+	ko *svcapitypes.ReplicationGroup,
+) {
+	//TODO: if in the future we need to set other annotations fields, ensure all get persisted
+	// as opposed to each making a copy of desired and only modifying one field
+
+	// copy desired annotations map into new map
+	desiredAnnotations := r.ko.ObjectMeta.GetAnnotations()
+	annotations := make(map[string]string)
+	for k, v := range desiredAnnotations {
+		annotations[k] = v
+	}
+
+	lastRequestedConfigs, err := json.Marshal(r.ko.Spec.LogDeliveryConfigurations)
+	if err != nil {
+		annotations[AnnotationLastRequestedLDCs] = "null"
```

Comment:
Is null unmarshaled and handled properly in this case? Also, if this is null, will it break the functionality of the controller?